inference/core/workflows/core_steps/utilities/dominant_color/v1.py

```diff
@@ -0,0 +1,119 @@
+to use in further processing or visualization tasks.
+
+Note: The block operates on the assumption that the input image is in RGB format. 
+"""
+
+
+class DominantColorManifest(WorkflowBlockManifest):
+    type: Literal["roboflow_core/dominant_color@v1", "DominantColor"]
+    model_config = ConfigDict(
+        json_schema_extra={
+            "name": "Dominant Color",
+            "version": "v1",
+            "short_description": SHORT_DESCRIPTION,
+            "long_description": LONG_DESCRIPTION,
+            "license": "Apache-2.0",
+            "block_type": "util",
+        }
+    )
+
+    image: Union[WorkflowImageSelector, StepOutputImageSelector] = Field(
+        title="Input Image",
+        description="The input image for this step.",
+        examples=["$inputs.image", "$steps.cropping.crops"],
+        validation_alias=AliasChoices("image", "images"),
+    )
+
+    @classmethod
+    def describe_outputs(cls) -> List[OutputDefinition]:
+        return [
+            OutputDefinition(name="rgb_color", kind=[LIST_OF_VALUES_KIND]),
+        ]
+
+    @classmethod
+    def get_execution_engine_compatibility(cls) -> Optional[str]:
+        return ">=1.0.0,<2.0.0"
+
+
+class DominantColorBlockV1(WorkflowBlock):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    @classmethod
+    def get_manifest(cls) -> Type[DominantColorManifest]:
+        return DominantColorManifest
+
+    def run(self, image: WorkflowImageData, *args, **kwargs) -> BlockResult:
+        np_image = image.numpy_image
+
+        # Downsample the image to speed up processing
+        height, width = np_image.shape[:2]
+        scale_factor = max(1, min(width, height) // 100)
+        np_image = np_image[::scale_factor, ::scale_factor]
+
+        pixels = np_image.reshape(-1, 3).astype(np.float32)
+
+        k = 4
+        max_iterations = 100
+        centroids = pixels[np.random.choice(pixels.shape[0], k, replace=False)]
+
+        for _ in range(max_iterations):
+            # Assign pixels to nearest centroid
+            distances = np.sqrt(((pixels[:, np.newaxis] - centroids) ** 2).sum(axis=2))
+            labels = np.argmin(distances, axis=1)
+
+            # Update centroids
+            new_centroids = np.zeros_like(centroids)
+            for i in range(k):
+                cluster_points = pixels[labels == i]
+                if len(cluster_points) > 0:
+                    new_centroids[i] = cluster_points.mean(axis=0)
+                else:
+                    # If cluster is empty, reinitialize to a random point
+                    new_centroids[i] = pixels[np.random.choice(pixels.shape[0])]
+
+            # Check for convergence
+            if np.allclose(centroids, new_centroids):
+                break
+
+            centroids = new_centroids
+
+        # Get the colors and their counts
+        colors = centroids
+        _, counts = np.unique(labels, return_counts=True)
```

Comment:
is `labels` variable initialised before first round of loop? if not lets do it or prevent `max_iterations` (since this will be param) from being <=0 - should be enough to put proper validation in manifest (something like `Field(gt=0)`)